docs/api/modulus.models.rst

```diff
@@ -2,6 +2,403 @@
+    ModulusModel = Module.from_torch(TorchModel, meta=MetaData())
+
+
+Saving and Loading Modulus Models
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As mentioned above, Modulus models are interoperable with PyTorch models. This means that
+you can save and load Modulus models using the standard PyTorch APIs however, we provide
+a few additional utilities to make this process easier. A key challenge in saving and
+loading models is keeping track of the model metadata such as layer sizes, etc. Modulus
+models can be saved with this metadata to a custom file ``.mdlus``. These files allow
+for easy loading and instantiation of the model. Below is an example of saving and
+instantiating a model from a ``.mdlus`` file.
+
+.. code:: python
+
+    >>> import torch
+    >>> from modulus.models import Module
+    >>> from modulus.models.mlp.fully_connected import FullyConnected
+    >>> model = FullyConnected(in_features=32, out_features=64)
+    >>> model.save("model.mdlus") # Save model to .mdlus file
+    >>> model.load("model.mdlus") # Load model weights from .mdlus file
+    >>> new_model = Module.from_checkpoint("model.mdlus") # Instantiate model from .mdlus file
+    >>> new_model
+    FullyConnected(
+     (layers): ModuleList(
+       (0): FCLayer(
+         (activation_fn): SiLU()
+         (linear): Linear(in_features=32, out_features=512, bias=True)
+       )
+       (1-5): 5 x FCLayer(
+         (activation_fn): SiLU()
+         (linear): Linear(in_features=512, out_features=512, bias=True)
+       )
+     )
+     (final_layer): FCLayer(
+       (activation_fn): Identity()
+       (linear): Linear(in_features=512, out_features=64, bias=True)
+     )
+   )
+
+.. note::
+   Inorder to make use of this functionality, the model must have json serializable
+   inputs to the ``__init__`` function. It is highly recommended that all Modulus
+   models be developed with this requirement in mind.
+
+
+Modulus Model Registry and Entry Points
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Modulus contains a model registry that allows for easy access and ingestion of
+models. Below is a simple example of how to use the model registry to obtain a model
+class.
+
+.. code:: python
+
+    >>> from modulus.registry import ModelRegistry 
+    >>> model_registry = ModelRegistry()
+    >>> model_registry.list_models()
+    ['AFNO', 'DLWP', 'FNO', 'FullyConnected', 'GraphCastNet', 'MeshGraphNet', 'One2ManyRNN', 'Pix2Pix', 'SFNO', 'SRResNet']
+    >>> FullyConnected = model_registry.factory("FullyConnected")
+    >>> model = FullyConnected(in_features=32, out_features=64)
+
+The model registry also allows exposing models via entry points. This allows for
+integration of models into the Modulus ecosystem. For example, suppose you have a
+package ``MyPackage`` that contains a model ``MyModel``. You can expose this model
+to the Modulus registry by adding an entry point to your ``setup.py`` file. For
+example, suppose your package structure is as follows:
+
+.. code:: python
```

Comment:
I would recommend converting this sample to use TOML instead of `setup.py` as new projects are [advised to avoid](https://setuptools.pypa.io/en/latest/userguide/quickstart.html#setup-py) `setup.py`.

Reply:
Agree! TOML is the way to go. We should probably also change how we do entrypoints in Modulus itself.

Reply:
Fixed this